extern/detray/CMakeLists.txt

```diff
@@ -13,7 +13,7 @@ message( STATUS "Building Detray as part of the TRACCC project" )
 
 # Declare where to get Detray from.
 set( TRACCC_DETRAY_SOURCE
-   "URL;https://github.com/acts-project/detray/archive/refs/tags/v0.95.0.tar.gz;URL_MD5;e5c89d7fc17f868eaf46e4eb3b8e630c"
+   "GIT_REPOSITORY;https://github.com/niermann999/detray;GIT_TAG;feat-more-assertions"
```

Comment:
The PR should really be marked as a draft until we can switch to a new Detray tag.

Reply:
Right, it is just that I also wanted to have the CI bridge run and see if the GPU code triggers any of the new assertions